### ISSUE TITLE
Add support for Encrypt fields

### ISSUE THREAD

**Issue by chris001** [salesagility/SuiteCRM#1755]
#### Issue

Add support for Encrypt fields, which was removed in SugarCRM CE 6.1 in 2011.

These encrypted fields are required by law in many countries, for storing identity information, such as, social security number, drivers license number, date of birth, credit card number, etc.

See forum request:
https://www.suitecrm.com/forum/developer-help/1740-securing-encrypting-a-custom-field
#### Expected Behavior

Go into Module Builder.  
You should be able to select field of type "Encrypt", label it "Social Security Number".
Add the field to your Quick Create, Detail and Edit views.
Save and click Deploy to make the the module active in your site.
Go into the module.
Create a new record.
Enter data into the Social Security Number field.
Save the new record.
Browse the database with phpMyAdmin.
Find the field for the new record you created.
The column for Social Security Number should be an illegible string of characters.
#### Actual Behavior

There is no "Encrypt" field type, since it was removed in SugarCRM CE 6.1 in 2011.
#### Possible Fix

I have created the fix and will file a pull request momentarily to have it pulled in.
.
#### Steps to Reproduce

1.Go into Module Builder.  
2. You should be able to select field of type "Encrypt", label it "Social Security Number".
3. There is no field of type "Encrypt".
#### Context

High priority.
SuiteCRM needs to comply with national laws requiring encrypted fields to safely contain personally identifying information, which is highly sought after by hackers and other unscrupulous people seeking to commit financial fraud.
#### Your Environment

- SuiteCRM Version used: 7.6.5
- Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Firefox
- Environment name and version (e.g. MySQL, PHP 7): MySQL 5.5, PHP 5.6
- Operating System and version (e.g Ubuntu 16.04): Centos 6.5

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Add support for Encrypt fields](https://trello.com/c/KDLKoHEu/86-add-support-for-encrypt-fields)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
